File: src/Sample.FSharp/MinimalZeroInstall.fs
open ZeroInstall.Backend
open ZeroInstall.Model
open ZeroInstall.Injector

let locator = new ServiceLocator(new CliHandler())
let solve = locator.Solver.Solve
let uncached = locator.SelectionsManager.GetUncachedImplementations
let fetch = locator.Fetcher.Fetch
let execute selections = (new Executor (selections, locator.Store)).Start()

let run requirements =
    let selections = solve requirements
    fetch (uncached selections)
    execute selections

[<EntryPoint>]
let main args =
    ignore(run(new Requirements(InterfaceID = args.[0])))
    0

File: src/Sample.CSharp/MinimalZeroInstall.cs
using ZeroInstall.Backend;
using ZeroInstall.Model;
using ZeroInstall.Injector;

class MinimalZeroInstall : ServiceLocator
{
    public static void Main(string[] args)
    {
        new MinimalZeroInstall().Run(new Requirements {InterfaceID = args[0]});
    }

    private MinimalZeroInstall() : base(new CliHandler())
    {}

    private void Run(Requirements requirements)
    {
        var selections = Solver.Solve(requirements);
        var missing = SelectionsManager.GetUncachedImplementations(selections);
        Fetcher.Fetch(missing);
        new Executor(selections, Store).Start();
    }
}
